R/gen2dist.r

```diff
@@ -11,17 +11,16 @@
 #'
 gen2dist <- function(vcf, biallelic = FALSE) {
 
-  if("vcfR" %in% rownames(utils::installed.packages()) == FALSE){
-    stop("gen2dist() requires the package vcfR")}
-  if (class(vcf)!='vcfR'){
-    stop("vcf must be of class 'vcfR'")
-  }
   #extract genotype information
-  gendat <- vcfR::extract.gt(vcf)
+  vcf_names <- names(vcf)
+  gendat <- vcf[(which(vcf_names=='FORMAT')+1):length(vcf_names)]
+  gendat <- sapply(gendat, substring, 1, 3)
+
   #swap slashes for pipes
   gendat <- sub('/','|',gendat)
-  #recode missing data as no mutation
-  gendat[is.na(gendat)] <- '0|0'
+  #recode NAs
+  gendat[is.na(gendat)] <- "0|0"
```

Comment:
This really relates to the old code, but I think if NAs are recoded, then there should be a warning saying it's happening - so just check if there are any and give a warning message if so...